model/project_vars.go

```diff
@@ -216,15 +221,36 @@ func (projectVars *ProjectVars) GetRestrictedVars() map[string]string {
 	return restrictedVars
 }
 
-func (projectVars *ProjectVars) GetAdminOnlyVars() map[string]string {
+// GetAdminOnlyVars will change to GetVars on EVG-16045 after removing restricted vars.
+func (projectVars *ProjectVars) GetAdminOnlyVars(t *task.Task) (map[string]string, error) {
 	adminOnlyVars := map[string]string{}
-
-	for k, v := range projectVars.Vars {
-		if projectVars.AdminOnlyVars[k] {
-			adminOnlyVars[k] = v
+	if evergreen.IsSystemActivator(t.ActivatedBy) {
```

Comment:
Wait sorry this isn't right for mainline commits; this just checks if the person who activated the task was not a person but was Evergreen. You want to check that the requester is in this list. https://github.com/evergreen-ci/evergreen/blob/f7284bd30cef2f5f12d976983406ee9e718e47bc/globals.go#L485 Sorry if I said the wrong thing.

Reply:
ahh okay